packages/flutter/test/widgets/image_test.dart

```diff
@@ -1225,7 +1358,33 @@ void main() {
   });
 }
 
-class TestImageProvider extends ImageProvider<TestImageProvider> {
+class ConfigurationAwareKey {
+  const ConfigurationAwareKey(this.provider, this.configuration)
+    : assert(provider != null),
+      assert(configuration != null);
+
+  final ImageProvider provider;
+  final ImageConfiguration configuration;
+
+  @override
+  int get hashCode => hashValues(provider, configuration);
+
+  @override
+  bool operator ==(Object other) {
+    return other is ConfigurationAwareKey
```

Comment:
operator == should compare runtimeType; see style guide

Reply:
Done.